netket/graph/lattice.py

```diff
@@ -12,25 +12,28 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from .graph import NetworkX
-from scipy.spatial import cKDTree
-from scipy.sparse import find, triu
-import numpy as _np
+from dataclasses import dataclass
 import itertools
-import networkx as _nx
-import warnings
-from typing import Tuple, Union, Optional
-from math import pi
 from itertools import product
-from dataclasses import dataclass
+from math import pi
+from typing import Sequence, Tuple, Union, Optional
+import warnings
+
+import networkx as _nx
+import numpy as _np
+from scipy.spatial import cKDTree
+from scipy.sparse import find, triu
 
 from netket.utils.semigroup import Identity, Element, PermutationGroup
 from netket.utils import HashableArray
+from netket.utils.types import Array
+
+from .graph import NetworkX
 
 tol_digits = 5
 cutoff_tol = _np.power(10.0, -tol_digits)
 """Tolerance for the maximum distance cutoff when computing the sparse distance matrix.
-This is necessary because of floating-point errors when computing the distance in non-trivial 
+This is necessary because of floating-point errors when computing the distance in non-trivial
 lattices.
 """
```

Comment:
See https://github.com/netket/netket/pull/702#discussion_r632394326.

Do we need those two values? Also, I think I missed it in the previous discussion, what was the reason this is a separate value instead of using `distance_atol` from `__init__` as for the nearest-neighbor search?

Reply:
This is related to the float → int "hashing logic", which is quite distinct from searching nearest neighbours. This logic will also have more general uses and will be moved from here soon, so I wouldn't tie it too hard to this class now...

Reply:
I agree with https://github.com/netket/netket/pull/702#discussion_r632394326 that having both variables is not a good thing, but neither will be around for too long, so let's not spend more brainpower on it

Reply:
I can live with that, just wanted to make sure.